test/e2e/network/netpol/test_helper.go

```diff
@@ -0,0 +1,145 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package netpol
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"sigs.k8s.io/yaml"
+)
+
+// prettyPrint a networkPolicy
+func prettyPrint(policy *networkingv1.NetworkPolicy) string {
+	raw, err := yaml.Marshal(policy)
+	framework.ExpectNoError(err, "marshal network policy to yaml")
+	return string(raw)
+}
+
+// CreateOrUpdatePolicy fails if it can not create or update the policy in the given namespace
+func CreateOrUpdatePolicy(k8s *Kubernetes, policy *networkingv1.NetworkPolicy, namespace string, isVerbose bool) {
+	if isVerbose {
+		framework.Logf("****************************************************************")
+		framework.Logf("Network Policy creating %s/%s \n%s", namespace, policy.Name, prettyPrint(policy))
+		framework.Logf("****************************************************************")
+	}
+
+	_, err := k8s.CreateOrUpdateNetworkPolicy(namespace, policy)
+	framework.ExpectNoError(err, "Unable to create/update netpol %s/%s", namespace, policy.Name)
+}
+
+// ValidateOrFail validates connectivity
+func ValidateOrFail(k8s *Kubernetes, model *Model, testCase *TestCase, isVerbose bool) {
+	ginkgo.By("Validating reachability matrix...")
+
+	// 1st try
+	ginkgo.By("Validating reachability matrix... (FIRST TRY)")
+	ProbePodToPodConnectivity(k8s, model, testCase)
+	// 2nd try, in case first one failed
+	if _, wrong, _, _ := testCase.Reachability.Summary(ignoreLoopback); wrong != 0 {
+		framework.Logf("failed first probe... retrying (SECOND TRY)")
+		ProbePodToPodConnectivity(k8s, model, testCase)
+	}
+
+	// at this point we know if we passed or failed, print final matrix and pass/fail the test.
+	if _, wrong, _, _ := testCase.Reachability.Summary(ignoreLoopback); wrong != 0 {
+		testCase.Reachability.PrintSummary(true, true, true)
+		framework.Failf("Had %d wrong results in reachability matrix", wrong)
+	} else {
+		if isVerbose {
+			testCase.Reachability.PrintSummary(true, true, true)
+		}
+		framework.Logf("VALIDATION SUCCESSFUL")
+	}
+}
+
+// UpdateNamespaceLabels sets the labels for a namespace
+func UpdateNamespaceLabels(k8s *Kubernetes, ns string, newNsLabel map[string]string) {
+	err := k8s.setNamespaceLabels(ns, newNsLabel)
+	framework.ExpectNoError(err, "Update namespace %s labels", ns)
+	err = wait.PollImmediate(1*time.Second, 30*time.Second, func() (done bool, err error) {
```

Comment:
nit, there are some constants defined allover the tests and the framework, but you can define your own ons
Check if you can normalize all the timeouts using constants, so if we want to modifu the test behavior we only need to change the value of the constant and not all the loops in the test

Reply:
👍

Reply:
I did a little grepping and didn't see anything obvious  to reuse here

Reply:
Let me clarify, there are some constants with timeouts in the e2e, but you can create your owns.
My point is that , you are using the same interval and timeout in the wait.PollImmediate() in several places. If it turns out we want to change one of both, we have to find all the wait. Loops and replace them, if you set a constant you only need to change the value in one place